### ISSUE TITLE
Can't install moduls via admin interface

### ISSUE THREAD

**Issue by simondueckert** [ether/etherpad-lite#5569]
I just updated my Debian server from version 10 to 11. I did a fresh install of etherpad-lite (with this tutorial: https://www.howtoforge.com/how-to-install-etherpad-on-debian-11/). I use my old mariadb database and the old settings.json. Node/Npm was also updated:

Node: v16.16.0
Npm: 8.11.0

When I try to install my needed plugins (e.g. adminpads2, headings2) via /admin/plugins the UI shows "installing" for a long time. When I reload the page the secion "Installed Plugins" is empty again.

Any ideas what that might be?

**Comment by muzzol** [comment]
it's the same for me. fresh git install on debian 11 and plugins fail to install from /admin page.

if I do it with commands it works ok BUT carefull following docs recomendation: **npm install --no-save --legacy-peer-deps ep_$plugin_name**

that fails for me on some plugins, so I just removed extra parameters and did it like usual:

**npm install  ep_themes**

also, remember you can easily restart your instance from **/admin/settings**

**Comment by JedMeister** [comment]
FWIW, I hit this same issue too.

The "fix" is to either use Node v14 (and everything will "just work") or install newer npm into Etherpad. Etherpad ships with npn v6 internally and uses that for installing plugins. In theory it should work ok, but in practice, it appears to cause issues and instability. Updating Etherpad's npm to v8 appears to resolve (most of) this issue. It will complain about deprecated switches, but otherwise appears to work fine.

**Comment by JedMeister** [comment]
@hikatanguy - I ended up just running Node 14. But I did confirm that updating the internal npm did solve the problem - at least on face value. I didn't "battle test" it.

I don't recall the exact steps, but I did a normal install (under Node 16) but once that was done, I cd'd into the Etherpad dir and somewhere in there (in a `bin` sub dir IIRC?) you'll find a `npm` executable. I then ran that to update itself. I'm not super familiar with Node and don't recall the details, but google says that the command should be something like:
```
./path/to/bin/npm install npm@8
```
Good luck!
